app/controllers/app.go

```diff
@@ -276,3 +276,45 @@ func (c App) SearchJSON(page int) revel.Result {
 	}
 	return c.RenderJSON(plogs)
 }
+
+func (c App) UploadJSON() revel.Result {
+	var uploadJSON models.JSONUploadOp
+	err := c.Params.BindJSON(&uploadJSON)
+	if err != nil {
+		revel.ERROR.Println("Error when binding JSON info", err)
+		return c.RenderJSON(models.JSONUploadResult{ false, 0, err.Error() })
+	}
+
+	// Check the shared secret
+	if uploadJSON.AuthToken != app.AuthToken {
+		return c.Forbidden("Invalid AuthToken")
+	}
+
+	plogData := uploadJSON.Upload
+
+	// Check input
+	_, err = GetUser(plogData.Protagonista)
+	// FIXME: Refactor if we need to add more checks like the ones below.
+	if err != nil {
+		revel.ERROR.Println("Error retrieving Protagonista", err)
+		c.Response.Status = http.StatusBadRequest
```

Comment:
I thought we decided a 400 was not in order here

Reply:
Apparently I didn't understand what the consensus was.

Reply:
Guys, please keep politics out of this.